design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,330 @@
+# TimeToValue: Wizard MVP Design
+
+## Description
+
+This design document will specify a possible solution for the time-to-value [wizard MVP](https://github.com/treeverse/lakeFS/issues/3411).  
+The wizard will provide a quick and clear way to start interacting with lakeFS.
+It will do that by allowing the users to
+1. Initialize a new repository with a given namespace (and name) with a ‘main’ default branch.
+2. Import their data (write metadata) into the new repository’s main branch by specifying the data’s S3 bucket namespace.
+3. Get custom Spark configurations and custom Hive metastore configurations to access lakeFS using the S3 Gateway (this is for the MVP).
+4. Summarise all actions performed (or skipped) in a README file which will be at the root of the initialized repository.
+
+---
+
+## System Overview
+
+![RouterFS with lakeFS URI](diagrams/wizard-mvp.png)
+[(excalidraw file)](diagrams/wizard-mvp.excalidraw)
+
+### Wizard UI Component
+
+The wizard UI component is responsible for the user’s Spark onboarding process. The process is as follows:
+1. Create a repository (named as the user wishes) and a ‘main’ branch in it.
+2. Import the user’s data to ‘main’ and display a progress bar (which will show a link to required permissions). Only available on cloud-deployed lakeFS.
+3. Generate Spark configurations: the wizard will return Spark configurations for the users to use in their Spark core-site.xml file, [databricks key-value format](https://docs.databricks.com/clusters/configure.html#spark-configuration), or [EMR JSON](https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html) ([core-site](https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html))
+   -> All three templates will be requested (each in a separate request).
+   1. Asks the user to enter the lakeFS endpoint (by default it will be https://`location.host`)
+   2. [Figma Design](https://www.figma.com/file/haD9y599LzW6LvsYBI2xWU/Spark-use-case?node-id=31%3A200)
+4. Generates a README file with all actions performed, and will commit it as a first commit (after the import) to the ‘main’ branch of the created repository.
+
+### Templating Service
+
+The [templating service](https://github.com/treeverse/lakeFS/pull/3373) is responsible for fetching, authenticating, and expanding the required templates and returning them to the client.  
+**Process**:  
+1. Get the template (the location should be specified in the incoming request). The file must be a valid [`html/template`](https://pkg.go.dev/html/template) or [`text/template`](https://pkg.go.dev/text/template) parsable template text.
```

Comment:
How do we distinguish the two?  I suggest the extension before `.tt`, say `PATH.something.tt` generates `PATH.something` and `PATH.html.tt` generates `PATH.html`.

Reply:
Cool